src/pages/content/listings.js

```diff
@@ -109,6 +110,29 @@ module.exports = {
 
             ctrl.fetch();
         };
+
+        ctrl.remove = function(data) {
+            var ref = db.child("content").child(ctrl.schema.key).child(data.key);
+
+            if(window.confirm("Remove " + data.name + "?")) {
+                ref.once("value", function(snap) {
+                    var data = snap.exportVal(),
+                        rev  = data.version || 1,
+                        dest = db.child("versions").child(snap.key()).child(rev);
+
+                    dest.set(data);
+                    ref.child("version").set(rev + 1);
```

Comment:
Setting the version data onto the ref you're about to remove seems a little weird.

The `remove` call should probably also happen within the value callback, otherwise it might delete while you're trying to move it.